### PR TITLE
Feature/get multiple items by title

### LINKED ISSUE #58: Add functionality to get multiple items by a title
## Summary

We want to enhance the way users can get items when using the Connect Node SDK. In this issue, we want to enable users to get a list of items that have the given title.

## Tasks to be done

- [x] Implement `listItemsByTitle` function in the `Items` class in `src/lib/resources.ts`. Important note: The API response returns an Item summary, not the full item, therefore once you get the list of item summaries, for each item you have to call `getItemById`.
   ```ts
   /**
    * Search for all Items with an exact match on title.
    *
    * @param {string} title
    * @returns {Promise<FullItem[]>}
    * @private
    */
   private async listItemssByTitle(
            vaultId: string,
            title: string,
        ): Promise<Response> {
        // functionality
   }
   ```
   Tip: You can move [this snippet](https://github.com/1Password/connect-sdk-js/blob/09bddd7f919373a5f5a8130ab332de5bc6024d1c/src/lib/resources.ts#L125-L136) from `get` to `listItemsByTitle` and then call `listItemsByTitle` in `get`. In this way, you avoid duplicate code in the two function.
- [x] Implement `listItemsByTitle` function in OPConnect class in `src/lib/op-connect.ts`:
  ```ts
  /**
   * Returns a list of Items with a matching Title value.
   *
   * @param {string} vaultId
   * @param {string} itemTitle
   * @returns {Promise<FullItem[]>}
   */
  public async listItemsByTitle(vaultId: string, itemTitle: string): Promise<FullItem[]> {
        return await this.items.listItemsByTitle(vaultId, itemTitle);
  }
  ```
- [x] Add a test for the new function in `__test__/op-connect.test.ts` below `get item` test case:
  ```ts
  test("list items by titte", async () => {
      // actual test here
  }
  ```

### PR DESCRIPTION
**Pull request by volodymyrZotov**
Resolves #58 

Added ability to get multiple items by title

### CODE REVIEW

File: __test__/op-connect.test.ts [hunk 1]
```diff
@@ -227,6 +227,49 @@ describe("Test OnePasswordConnect CRUD", () => {
             expect(vault.name).toEqual(vaultTitle);
         });
     });
+
+    describe("list items by title", () => {
+        const title = "some title";
+        const op = OnePasswordConnect(testOpts);
+        const getItemsByTitleMock = (title: string) => nock(mockServerUrl)
```
Comment (edif2008) [inline __test__/op-connect.test.ts:234]: This is a really nice code simplification. Do you think you can do the same thing for vaults tests? (we can do this in a separate PR)

File: src/lib/resources.ts [hunk 2]
```diff
@@ -159,6 +159,26 @@ export class Items extends OPResource {
         );
     }
 
+    /**
+     * Search for all Items with exact match on title.
```
Comment (edif2008) [inline src/lib/resources.ts:163]: Tiny nitpick (to match other doc formatting):
```suggestion
     * Search for all Items with exact match on Title.
```

**Review by edif2008 — COMMENTED** [review]
Functional review: ✅ 
Works as expected 😄 

Code review: ✅ 
Nice touches in terms of code simplification especially in the tests. All looks clear and on point. 😄

File: __test__/op-connect.test.ts (hunk 1 above)
Comment (volodymyrZotov) [reply __test__/op-connect.test.ts:234]: Yes, we even can extract everything in some `api.mock.ts` module and use it in the tests in this way. Let's do such a refactoring in the other PR :)

File: src/lib/resources.ts (hunk 2 above)
Comment (volodymyrZotov) [reply src/lib/resources.ts:163]: fixed

File: __test__/op-connect.test.ts [hunk 3]
```diff
@@ -227,6 +227,49 @@ describe("Test OnePasswordConnect CRUD", () => {
             expect(vault.name).toEqual(vaultTitle);
         });
     });
+
+    describe("listItemsByTitle", () => {
+        const title = "some title";
+        const op = OnePasswordConnect(testOpts);
```
[the remark this answers is not in the document]
Comment (volodymyrZotov) [reply __test__/op-connect.test.ts:233]: removed

**Review by kpcraig — APPROVED** [review]
